music/src/core/melodies.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Module containing functions for converting to and from quantized melodies.
+ *
+ * @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Imports
+ */
+import {INoteSequence, NoteSequence} from '../protobuf/index';
+
+import * as sequences from './sequences';
+
+export const NO_EVENT = 0;
+export const NOTE_OFF = 1;
+const FIRST_PITCH = 2;
+
+/**
+ * Melody representation with note onsets, `NO_EVENT` (sustain) and `NOTE_OFF`.
+ *
+ * @param events An array of melody events.
+ * @param minPitch The minimum pitch to represent.
+ * @param maxPitch The maximum pitch to represent.
+ *
+ */
+export class Melody {
+  readonly events: Int32Array;
```

Comment:
is that the same as `Number[]` ?

Reply:
I think it is not even though I have forgotten the small amount of TypeScript I knew.  I tried using a number[] but the tf.tensor function didn't want to take one of those (in data.py) so to avoid multiple array conversions I just used Int32Array.  Is that weird?